src/js/controller/controller.js

```diff
@@ -244,6 +244,16 @@ Object.assign(Controller.prototype, {
 
             _model.change('playlistItem', function(model, playlistItem) {
                 if (playlistItem) {
+                    const { title, image } = playlistItem;
+                    if ('mediaSession' in navigator && (title || image)) {
```

Comment:
This code comes right before we trigger "playlistItem". Any exceptions would break player functionality. As such you need to add better guards here.

Just because "mediaSession" is in window, that does not guarantee that "MediaMetadata" is as well. Add it to the if block.

I would also like to see the following code in a `try catch` block to handle any potential exceptions instantiating `MediaMetadata`. For example if artwork src is not parsed correctly a `TypeError` will be thrown (https://wicg.github.io/mediasession/#convert-artwork-algorithm).

Don't we also need action handlers to control the player from the device's info panel?
 https://developer.mozilla.org/en-US/docs/Web/API/MediaSession